### ISSUE TITLE
Fall Guys - Old banner in library & missing files error

### ISSUE THREAD

**Issue by FireFox1121** [Heroic-Games-Launcher/HeroicGamesLauncher#2696]
### Describe the bug

When I started launcher, I still see old banner logo in my library, and after Fall Guys update (then reinstall), i got missing files error.

### Add logs

```text
(16:29:34) INFO:    [Legendary]:        Refreshing library...
(16:29:34) INFO:    [Legendary]:        Refreshing Epic Games...
(16:29:34) INFO:    [Connection]:       Connectivity: check-online
(16:29:34) INFO:    [Connection]:       Pinging external endpoints
(16:29:34) INFO:    [Legendary]:        Running command: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(16:29:34) INFO:    [Legendary]:        Legendary location: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(16:29:34) INFO:    [Gog]:              GOGDL location: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:29:34) INFO:    [Gog]:              GOGDL location: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(16:29:34) INFO:    [Legendary]:        Game list updated, got 4 games & DLCs
(16:29:34) WARNING: [Backend]:          Protocol already registered.
(16:29:34) INFO:    [Gog]:              Running command: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/computer/.config/heroic/gog_store/auth.json --version
(16:29:34) INFO:    [Connection]:       Connectivity: online
(16:29:35) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(16:29:35) INFO:    [Backend]:          

Heroic Version: 2.7.1 Eustass Kid
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.1

Electron Version: 23.1.4
Chrome Version: 110.0.5481.192
NodeJS Version: 18.12.1

OS: LinuxMint KERNEL: 5.15.0-71-generic ARCH: x64
CPU: AMD Ryzen 7 4700G with Radeon Graphics @3.6 GOVERNOR: performance
RAM: Total: 15 GiB Available: 11.99 GiB
GRAPHICS: GPU0: NVIDIA GeForce GTX 1050 Ti VRAM: 4096MB DRIVER: 525.105.17GPU1: Renoir VRAM: 512MB 
PROTOCOL: x11

(16:29:35) INFO:    [Legendary]:        Running command: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(16:29:36) INFO:    [Backend]:          Loading Screen Ready
(16:29:36) INFO:    [Frontend]:         Refreshing Library
(16:29:36) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(16:29:36) INFO:    [Gog]:              Found 0 game(s) to update
(16:29:36) INFO:    [Backend]:          Frontend Ready
(16:29:36) INFO:    [ProtocolHandler]:  received 'heroic://launch/legendary/0a2d9f6403244d12969e11da6713137b'
(16:29:36) INFO:    [Backend]:          Launching Fall Guys (0a2d9f6403244d12969e11da6713137b)
(16:29:36) INFO:    [Backend]:          Preventing display from sleep
(16:29:36) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-latest
(16:29:36) INFO:    [Backend]:          Preventing machine to sleep
(16:29:36) INFO:    [Backend]:          Stopping Power Saver Blocker
(16:29:36) INFO:    [Backend]:          Checking for new Heroic Updates
(16:29:37) INFO:    [Legendary]:        Checking for game updates: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(16:29:37) INFO:    [Winetricks]:       Downloading Winetricks
(16:29:37) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-latest
(16:29:37) WARNING: [Backend]:          You are using Proton, this can lead to some bugs. Please do not open issues with bugs related to games
(16:29:37) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-latest
(16:29:37) INFO:    [Backend]:          Checking if wine version exists: Proton - Proton-GE-latest
(16:29:37) DEBUG:   [Backend]:          Running Wine command: run wineboot --init
(16:29:38) INFO:    [Legendary]:        Found 0 games to update
(16:29:40) INFO:    [Legendary]:        Running command: /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 0a2d9f6403244d12969e11da6713137b --json --offline
(16:29:40) INFO:    [Legendary]:        Launching Fall Guys: STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/computer/.steam/steam STEAM_COMPAT_DATA_PATH="/mnt/sdb1/Games/Heroic/Prefixes/Fall Guys" PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 PROTON_EAC_RUNTIME=/home/computer/.config/heroic/tools/runtimes/eac_runtime STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId=heroic-FallGuys PROTON_LOG_DIR=/home/computer /usr/local/bin/mangohud --dlsym /home/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 0a2d9f6403244d12969e11da6713137b --language en --no-wine --wrapper "'/home/computer/.config/heroic/tools/proton/Proton-GE-latest/proton' run" --skip-version-check
(16:29:41) INFO:    [Backend]:          Starting the Download Queue
(16:35:19) INFO:    [Backend]:          Stopping Display Power Saver Blocker
```

### Steps to reproduce

Reinstalling the game and reinstalling VC++ Redist won't help.
I also tried another user profile with clean data and config, but it not helps.

### Expected behavior

No response.

### Screenshots

![Screenshot_20230510_164342](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/f345d1c0-4a77-4fb2-bb9c-6dc829ff7fe7)
![Screenshot_20230510_155744](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/b1f3369f-aab4-401f-8b64-99af65a20dd8)

### Heroic Version

Latest Stable

### System Information

Ubuntu 20.04 LTS (Mint 20.2)

### Additional information

Wine - Proton-GE-Latest
Game - Fall Guys Creative Update

**Comment by hardpenguin** [comment]
I can reproduce this issue as well on Debian Sid. Missing files error is present.

Although I use "Wine-GE-Proton7-22" for running the game, in line with this guide: https://boilingsteam.com/fall-guys-install-free-linux/

It is interesting though that the issue affects Windows version of the Heroic launcher as well ☺️

**Comment by stogle** [comment]
Same issue on Steam Deck.

I also got the same Missing Files Error with EGS on Windows, as did [some](https://www.reddit.com/r/FallGuysGame/comments/13dqz9i/fall_guys_missing_files_error_after_new_season/) [others](https://www.reddit.com/r/FallGuysGame/comments/13dlrej/missing_files_after_downloading_update_ive_tried/). This [tweet](https://twitter.com/fallguysowl/status/1656288264775430150?s=46&t=wD9o3NiZtN9N9WBOPZNwYA) led me to the [Missing Game Files or 200_19002 Error](https://www.epicgames.com/help/en-US/fall-guys-c9505712380315/technical-support-c9505744757531/missing-game-files-or-200-19002-error-a9505717140891) help page, and reinstalling Epic Online Services as describe [here](https://www.epicgames.com/help/en-US/fall-guys-c9505712380315/technical-support-c9505744757531/how-to-fix-fall-guys-crash-and-technical-issues-on-pc-a9505718402203#h_01GJFTCMHKBG5DDT7Y9E353R9P) fixed the issue. Trying to update/reinstall the EOS Overlay in Heroic didn't help though.

**Comment by FireFox1121** [comment]
I rechecked game files from Steam (clean files), and some of files are really broken/not updated:
![Screenshot_20230510_205813](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/3daac901-b75e-4661-b9ee-008f897c7ffa)
![Screenshot_20230510_205822](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/48416242-734a-4150-ad28-d2e5b8a2ef79)
![Screenshot_20230510_205836](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/d4668d7a-1159-46ca-8d9c-24ff36e7171d)

**Comment by FireFox1121** [comment]
### This is EOS Overlay error (temp fix)
Just delete EOSSDK-Win64-Shipping.dll from FallGuys_client_game_Data/Plugins/x86_64 and this error **_is gone_**.

**Comment by FireFox1121** [comment]
Fall Guys - Player log, if delete EOSSDK or place EOSSDK from Epic Dev website:

```
Exception: NULL EOS Platform returned by native code: issue probably occurred in GFX Plugin!
  at PlayEveryWare.EpicOnlineServices.EOSManager+EOSSingleton.GetEOSPlatformInterface () [0x00000] in <00000000000000000000000000000000>:0 
  at EOSUtilities.PauseOverlay (System.Boolean pause) [0x00000] in <00000000000000000000000000000000>:0 
  at EOSManagerFG.Update () [0x00000] in <00000000000000000000000000000000>:0 
MPG.Utility.CustomLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
 
(Filename: currently not available on il2cpp Line: -1)
```

**Comment by hardpenguin** [comment]
> ### This is EOS Overlay error (temp/not full fix)
> 
> Just delete EOSSDK-Win64-Shipping.dll from FallGuys_client_game_Data/Plugins/x86_64 and this error **_is gone_**.
> 
> UPD: I'm stuck at loading screen

Good job investigating but yeah, I am also stuck in the loading screen after deleting this file.

I don't think the issue would be with this file? It has not been updated recently for me.

**Comment by hardpenguin** [comment]
Here is another solution that I am willing to give a try: https://www.reddit.com/r/FallGuysGame/comments/13dn570/missing_files_on_steam_deck/jjqmh9w/?context=3

**Comment by FireFox1121** [comment]
> Here is another solution that I am willing to give a try: https://www.reddit.com/r/FallGuysGame/comments/13dn570/missing_files_on_steam_deck/jjqmh9w/?context=3

Dude, it worked, thank you

I'll place it here:
```
Hey! I finally managed to get it working on the Steam Deck!

From the previous season, I followed the instructions from this video: https://youtu.be/exZ9T7eZlZ0

If you sort the comments by newest first, the guy mentions how to fix it. You can still sort of follow the video, but with some changes. But the gist is:

    Uninstall the game and the Wine prefix

    Reinstall again and use Proton 8-2 instead of 7-26

    Select RunFallGuys.exe instead of FallGuys_game_client.exe as the executable

    In the game’s settings, select the “Run exe on Prefix” button and choose the EpicOnlineServicesInstaller.exe file

    When the game starts for the first time, there will be a Microsoft redistributable program thingy that needs to be installed

There’s a step in the video to create a symlink for the easyanticheat_x64.so file, but I skipped that since it was already in the folder. It should then work.

My only gripe is that there’s a console or MS-DOS-like screen that first appears when you start the game. It just looks kinda unsightly. hahaha
```

**Comment by stogle** [comment]
>     Select RunFallGuys.exe instead of FallGuys_game_client.exe as the executable
> 
>     In the game’s settings, select the “Run exe on Prefix” button and choose the EpicOnlineServicesInstaller.exe file

These are the two key steps. It should not be necessary to uninstall and reinstall the game first.

**Comment by Laitinlok** [comment]
It doesn't work for me unfortunately

**Comment by hardpenguin** [comment]
> > ```
> > Select RunFallGuys.exe instead of FallGuys_game_client.exe as the executable
> > 
> > In the game’s settings, select the “Run exe on Prefix” button and choose the EpicOnlineServicesInstaller.exe file
> > ```
> 
> These are the two key steps. It should not be necessary to uninstall and reinstall the game first.

Yes, I can confirm that 👍️

**Comment by SteavenGamerYT** [comment]
It doesn't work for me unfortunately

**Comment by hardpenguin** [comment]
You guys must be doing something wrong.

Make sure to force the update of the game first. Then set up RunFallGuys.exe as alternative executable. And then "Run exe on prefix" with EpicOnlineServicesInstaller that you will find in the Fall Guys folder.

If that fails, perhaps it might make sense to uninstall, clean prefix and configuration files, and start anew.

**Comment by FireFox1121** [comment]
I just started from fresh prefix and selected RunFallGuys.exe EXE in Heroic.
Still works.

My settings:
![Screenshot_20230516_150917](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/7e7949ee-6f02-49f2-bb40-609983a81935)
![Screenshot_20230516_150930](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/e39415de-d021-40ce-a028-a7465e038fed)

**Comment by FireFox1121** [comment]
Second account. All works fine.
![Screenshot_20230516_160903](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/4268cea2-a6e9-4aee-902a-4840a51e3120)
![Screenshot_20230516_161054](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/94104868/0dcae986-12a4-4f63-8104-81a630f58ba6)

**Comment by FireFox1121** [comment]
How I fixed this error:

1. Remove WINEPREFIX for this game.
2. Click "Verify and Repair" (or re-download game by uninstall and install)
3. Click "Enable EOS Overlay"
4. In settings click "Run EXE on PREFIX" and select file from "game_dir/EpicOnlineServices/EpicOnlineServicesInstaller.exe"
4. Wait 2-3 minutes for complete EOS installation. (Or check logs)
5. Select alt EXE RunFallGuys.exe and run game. Console opens and EOS gonna be updated.
6. EAC will shown with transparent banner, after this game gonna started and you can play.

**Comment by CoolPenguin27** [comment]
for some reason, none of these fixes work for me, even after attempting the steps multiple times and switching wine versions. (Kubuntu 23.04, Flatpak)

**Comment by Stoneblackdog** [comment]
Doesn't work for me either. Arch Linux, AUR bin version.

**Comment by carlosmax3D** [comment]
I tried several times and It doesn't work, The only thing that worked is the installer with lutris, using the EpicGameLauncher, also I noticed that if I install the prefix on a BTRFS filesystem, the game pops the message of missing files, but if the prefix was on a EXT4 partition and the game files was on a BTRFS filesystem, works just fine.

**Comment by Laitinlok** [comment]
Tried lutris, still not working :(

**Comment by FireFox1121** [comment]
About old banner:
Delete ~/.config/legendary with closed Heroic, open Heroic and new banner appeared.

**Comment by flavioislima** [comment]
> About old banner: Delete ~/.config/legendary with closed Heroic, open Heroic and new banner appeared.

## THIS WILL ERASE YOUR EPIC GAMES LOGIN AND INSTALLED GAMES INFORMATION

Instead of deleting the whole folder you can delete only the metadata folder inside it and clean the heroic cache on settings. Then restart Heroic and wait for the games to refresh.

**Comment by CoolPenguin27** [comment]
I somehow got it working on a new install of kubuntu 23.04 (flatpak) (prev install was depthboot)

**Comment by matteobtt** [comment]
It is no longer necessary to set the alternative executable to `RunFallGuys.exe` because the override at https://api.legendary.gl/v1/version.json has been removed, but you have to force the game update (Right click > Details > three dots in the upper right corner > Force Update if Available) or edit the `.config/legendary/installed.json` file (for Fltapak the file is `~/.var/app/com.heroicgameslauncher.hgl/config/legendary/installed.json`).

**Comment by queenbiscuit311** [comment]
Worked on exactly one launch for me, then hasn't worked since.

**Comment by giopalma** [comment]
@Stoneblackdog worked for me two times, now it doesn't work anymore (Steam version)

**Comment by krompus** [comment]
I had the missing files error on a fresh Steam install, deleting my prefix, relaunching the game and pressing Cancel on the second Epic Online Services popup (the 87ish MB one) lets me launch the game and play without problems. Have to Cancel that popup every time or it'll give the Missing Files error. Would appreciate it if someone knows how to fix this so I don't have to click Cancel every time.

**Comment by siggiboy81** [comment]
This step-by-step guide explains to install Fall Guys from scratch on Ubuntu based on the newest version of Fall Guys. Lots of information from videos and information in this github has been used. Tested it on two computers and worked on both, so hope it will help others as well: https://www.ipaddressguide.org/how-to-install-and-play-fall-guys-on-ubuntu/

**Comment by giopalma** [comment]
@krompus workaround also works for me. The second update of EOS breaks the game, without doing it (pressing Cancel when the second update starts) I can play the game without problems

**Comment by Angeletjd** [comment]
a month ago i did all of this and it worked just fine and then there was an update yesterday and now nothing is working anymore
when i start the game half of my screen goes black and thats it....
i uninstalled the game and followed all the steps again but when i try to do
 "In the game’s settings, select the “Run exe on Prefix” button and choose the EpicOnlineServicesInstaller.exe file"
nothing happens 
it wont start the exe no matter which wine prefix i use and how long i wait
does anyone have an idea why?

**Comment by giopalma** [comment]
> "In the game’s settings, select the “Run exe on Prefix” button and choose the EpicOnlineServicesInstaller.exe file"
nothing happens

Nothing happens, it's normal. Go in this folder `~/Games/Heroic/Prefixes/default/Fall Guys/drive_c/ProgramData/Epic/EpicOnlineServices/EOSInstaller/Logs`

And check if you have a file like `EOSInstaller-XXXXXX.log` and it's filled.

Also, if you have problems, try to delete the prefix folder `~/Games/Heroic/Prefixes/default/Fall Guys`

Use `RUN EXE ON PREFIX` to run EpicOnlineServicesInstaller.exe and wait some seconds (you can check the file I said before)

EOS Installer will pop up, wait the end of the first update (around 140MB) and cancel the second update (around 84MB). You must do this every time you start the game. This is the only workaround we found, thanks to @krompus

**Comment by queenbiscuit311** [comment]
just tried this, still absolutely nothing

**Comment by queenbiscuit311** [comment]
never mind, after retrying a few times, the workaround shown by @krompus and @Anuskuss seem to work. Just have to click cancel every time on the one pop up.

**Comment by queenbiscuit311** [comment]
after retrying a few more times, it seems that basically you have to keep reinstalling epic online services and cancelling the install until it decides to stop trying to update it and you dont have to click cancel anymore. 

I dont know if its what caused it to work, but there was a specific thing I did before I got it to work reliably. 

I got the game to load once with the friends list working. I then closed it and removed my users permission to write to the wine prefix epic games folder with a simple sudo chown -R root:root "~/Games/Heroic/Prefixes/default/Fall Guys/drive_c/Program Files (x86)/Epic Games/", ran the game and terminated it twice since it wont load without write access to that folder, then gave my user account ownership again. After running the game once again, it worked fine and it hasn't bothered me with trying to update epic online or given me the missing files error since. ill see if it sticks.

**Comment by Arnav-lunatic** [comment]
Check Out -
https://www.reddit.com/r/FallGuysGame/comments/13dn570/missing_files_on_steam_deck/

**Comment by queenbiscuit311** [comment]
Here's the solution that works for me:

Set 
DOTNET_BUNDLE_EXTRACT_BASE_DIR="" in your environment variables 

delete your wine prefix

run the eos installer and install both dependencies. do not cancel any of them on game startup.

Special thanks to _dirtblock on the heroic discord for this, works perfectly for me.

**Comment by mibofra** [comment]
> Here's the solution that works for me:
> 
> Set DOTNET_BUNDLE_EXTRACT_BASE_DIR="" in your environment variables
> 
> delete your wine prefix
> 
> run the eos installer and install both dependencies. do not cancel any of them on game startup.
> 
> Special thanks to _dirtblock on the heroic discord for this, works perfectly for me. it seems to be the most reliable one by far.
> 
> you can search for the forum post I made about this on the heroic discord if you want to see the original thread.

Confirming it seems to work!

**Comment by Etaash-mathamsetty** [comment]
fixed in next release: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/2990

**Comment by felipecrs** [comment]
Jesus Christ. Me and my friend spent the whole night trying to make this work as of today with latest GE-Proton and Heroic.

After trying inumerous things, what worked to fix the "missing files" was to run the EpicOnlineServices.exe inside the prefix.

Apparently, "Enable EOS" and installing EOS in the main menu of Heroic did not make any difference and EOS was not installed after all in the prefix.

**Comment by felipecrs** [comment]
Just to confirm, here is what currently takes to get Fall Guys working with Heroic.

1. Download game
2. Set Wine-GE-Proton latest or Proton-GE-Proton latest, both works fine
3. Enable Easy Anti cheat support in game options within Heroic
4. Go to the game Wine Options, select Run Exe in Prefix and select the Heroic/Games/Fall Guys/Epic onlineServices/Epic onlineServices installer.exe and wait until the button stops blinking.
5. Run the game. Will work!

Note, you don't need:

1. To install EOS Overlay on main Heroic settings
2. To enable EOS Overlay in Heroic game settings
3. To set any alternative exe
4. To move any .so file around
5. To set any launch option
6. To set any environment variable
7. To change the .ini file to point to a different exe

That said, is there any chance Heroic can fix number 4 so that such tricky (for those who didn't know) thing is not required? (EpiconlineservicesInstaller.exe)

**Comment by arielj** [comment]
we have this wiki guide https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/Fall-Guys

like time I tested the game you only need those steps (and in the next version of heroic it will just work out of the box because heroic will install the epic online services automatically)

**Comment by felipecrs** [comment]
Omg that's awesome. If only I had found such wiki page.

And it's great to know that such step won't be required in the future.

Maybe automatic enablement of easy anti cheat support can also happen in the future, considering that AreWeAntiCheatYet provides the information on whether the game used Easy or not.

**Comment by Etaash-mathamsetty** [comment]
> Omg that's awesome. If only I had found such wiki page.
> 
> And it's great to know that such step won't be required in the future.
> 
> Maybe automatic enablement of easy anti cheat support can also happen in the future, considering that AreWeAntiCheatYet provides the information on whether the game used Easy or not.

it should be enabled automatically for every game, it should be enabled unconditionally. this is a bug, will look into it soon

**Comment by graves501** [comment]
> we have this wiki guide https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/Fall-Guys
> 
> last time I tested the game you only need those steps (and in the next version of heroic it will just work out of the box because heroic will install the epic online services automatically)

Can confirm that the steps in the wiki guide work as of June 2024. I used to use the `Proton GE` version, but I had to switch to `Wine GE` like stated in the guide to make the game work again.

**Comment by mibofra** [comment]
Hello! Maybe it just me, but last time I tried to start again Fall Guys, it did not work. Tried re-installing (deleting the wine prefix, logs and confs, everything could be deletable) a bit more of a dozen of times, ending setting `DOTNET_BUNDLE_EXTRACT_BASE_DIR=""` again in the env, MANUALLY, and it works as expected.

So, this is more a reminder to check if https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/2990 is still applied or something (other modifications to Heroic/other merge requests) ovewritten the default config/the default config for the specific app in the meanwhile. Thanks :)
